mathics/eval/strings.py

```diff
@@ -1,13 +1,241 @@
+"""
+String-related evaluation functions.
+"""
+import re
+from binascii import hexlify
+from typing import Optional, Tuple
+
 from mathics.core.atoms import String
 from mathics.core.element import BaseElement
 from mathics.core.evaluation import Evaluation
+from mathics.core.expression import Expression
 from mathics.core.symbols import Symbol
+from mathics.core.systemsymbols import (
+    SymbolBlank,
+    SymbolDigitCharacter,
+    SymbolEndOfLine,
+    SymbolEndOfString,
+    SymbolHexadecimalCharacter,
+    SymbolLetterCharacter,
+    SymbolNumberString,
+    SymbolStartOfLine,
+    SymbolStartOfString,
+    SymbolWhitespace,
+    SymbolWhitespaceCharacter,
+    SymbolWordBoundary,
+    SymbolWordCharacter,
+)
 from mathics.eval.makeboxes import format_element
 
+_regex_longest = {
+    "+": "+",
+    "*": "*",
+}
+
+_regex_shortest = {
+    "+": "+?",
+    "*": "*?",
+}
+
+# Regular expressions for various symbols.
+# Note: the regexp patterns below must not contain
+# re global flag like ?u or ?i.
+REGEXP_FOR_SYMBOLS = {
+    SymbolNumberString: r"[-|+]?(\d+(\.\d*)?|\.\d+)?",
+    SymbolWhitespace: r"\s+",
+    SymbolDigitCharacter: r"\d",
+    SymbolWhitespaceCharacter: r"\s",
+    SymbolWordCharacter: r"[^\W_]",
+    SymbolStartOfLine: r"^",
+    SymbolEndOfLine: r"$",
+    SymbolStartOfString: r"\A",
+    SymbolEndOfString: r"\Z",
+    SymbolWordBoundary: r"\b",
+    SymbolLetterCharacter: r"[^\W_0-9]",
+    SymbolHexadecimalCharacter: r"[0-9a-fA-F]",
+}
+
+
+def _encode_pname(name):
+    return "n" + hexlify(name.encode("utf8")).decode("utf8")
 
+
+# A better thing to do would be to write a pymathics module that
 def eval_ToString(
     expr: BaseElement, form: Symbol, encoding: String, evaluation: Evaluation
 ) -> String:
     boxes = format_element(expr, evaluation, form, encoding=encoding)
     text = boxes.boxes_to_text(evaluation=evaluation)
     return String(text)
+
+
```

Comment:
Maybe this stuff belongs to a module `mathics.core.convert.regex` instead of `mathics.eval.string`

Reply:
Yes, I think that is better. I was thinking that too.